atlasaction/action.go

```diff
@@ -118,22 +126,41 @@ func MigrateLint(ctx context.Context, client *atlasexec.Client, act *githubactio
 		Context:   string(buf),
 		Web:       true,
 		Writer:    &resp,
+		Format:    `{"URL": "{{ .URL }}", "DiagnosticsCount": {{ .DiagnosticsCount }}, "FilesCount": {{ len .Files }}}`,
 	})
-	url := strings.TrimSpace(resp.String())
-	if url != "" {
-		act.SetOutput("report-url", url)
-		if publishErr := publishResult(url, err, act); publishErr != nil {
-			act.Warningf("unable to publish lint report: %v", publishErr)
-		}
+	if err != nil && !errors.Is(err, atlasexec.LintErr) {
+		return err
 	}
-	return err
+	if err := json.NewDecoder(&resp).Decode(&payload); err != nil {
+		return fmt.Errorf("decoding payload: %w", err)
```

Comment:
Is this error user facing?

Reply:
What happens when there is an error that isn't due to linting? For example: bad url format for dev-url

Reply:
yes, though it shouldn't happen since it's not dependent on user input (i.e it will happen if Atlas doesn't return a JSON output). if it does, it will be helpful in a user report

Reply:
> What happens when there is an error that isn't due to linting? For example: bad url format for dev-url

If the error isn't due to linting, the code exits here. The action will fatally fail with the error message.

Reply:
Right, I see. That would be printed to stderr and it's not a linting error